Kernel/Process.cpp

```diff
@@ -1202,6 +1202,14 @@ int Process::sys$chdir(const char* path)
     return 0;
 }
 
+int Process::sys$fchdir(int fd)
+{
+    auto* description = file_description(fd);
+    if (!description)
+        return -EBADF;
+    return sys$chdir(description->absolute_path().characters());
```

Comment:
Instead of simply forwarding to `sys$chdir`, this should take advantage of the already-resolved `Custody` in the referenced `FileDescription`.
Also, we need to check if the description, otherwise fail with `ENOTDIR`.
See POSIX spec for `fchdir()`: https://pubs.opengroup.org/onlinepubs/9699919799/functions/fchdir.html

(Note: A `Custody` is a sort-of cache entry that describes the relationship between a parent inode and one of its children.)

```cpp
if (!description->is_directory())
    return -ENOTDIR;
m_cwd = description->custody();
```